auto_tests/save_friend_test.c

```diff
@@ -31,18 +31,33 @@ static void set_random(Tox *m, bool (*setter)(Tox *, const uint8_t *, size_t, To
     setter(m, text, SIZEOF_VLA(text), nullptr);
 }
 
+static void alloc_string(uint8_t **to, size_t length)
+{
+    if (*to != nullptr) {
```

Comment:
`free(nullptr)` is fine, so this is a bit odd.

Reply:
Good point. I wasn't thinking about that, I thought "if it's already set, it must be freed".